src/OpenTelemetry.AutoInstrumentation.StartupHook/RulesEngine/RuleEngine.cs

```diff
@@ -0,0 +1,67 @@
+// <copyright file="RuleEngine.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using OpenTelemetry.AutoInstrumentation.Logging;
+
+namespace OpenTelemetry.AutoInstrumentation.RulesEngine;
+
+internal class RuleEngine
+{
+    private static readonly IOtelLogger Logger = OtelLogging.GetLogger("StartupHook");
+
+    private readonly List<Rule> _rules = new()
+    {
+        new OpenTelemetrySdkRule
+        {
+            Name = "OpenTelemetry SDK Validator",
+            Description = "Ensure that the OpenTelemetry SDK version is not older than the version used by the Auto-Instrumentation"
```

Comment:
I think that it could be declares in Rule classes (together with abstract class contract change).